### PR TITLE
Report dtos ii

### PR DESCRIPTION
**Pull request by jlhogan**

### CODE REVIEW

File: src/Hedwig/Models/Reports/CdcReport/CdcReportDTO.cs [hunk 1]
```diff
@@ -0,0 +1,17 @@
+using System;
+using System.Collections.Generic;
+using Hedwig.Validations;
+
+namespace Hedwig.Models
+{
+	public class CdcReportDTO
+	{
+		public int Id { get; set; }
+		public FundingSource Type { get; set; }
```
Comment (cailyncodes) [inline src/Hedwig/Models/Reports/CdcReport/CdcReportDTO.cs:10]: Is this needed?

File: src/Hedwig/Models/Reports/OrganizationReport/OrganizationReportSummaryDTO.cs [hunk 2]
```diff
@@ -0,0 +1,13 @@
+using System;
+
+namespace Hedwig.Models
+{
+	public class OrganizationReportSummaryDTO
+	{
+		public int Id { get; set; }
+		public FundingSource Type { get; set; }
```
Comment (cailyncodes) [inline src/Hedwig/Models/Reports/OrganizationReport/OrganizationReportSummaryDTO.cs:8]: Likewise here

File: test/HedwigTests/Filters/ValidateEntityFilterTests.cs [hunk 3]
```diff
@@ -15,64 +15,7 @@ namespace HedwigTests.Filters
...
-			_validator.Verify(v => v.Validate(It.IsAny<INonBlockingValidatableObject>(), null), times);
-		}
-
-		[Theory]
-		[InlineData(false)]
-		[InlineData(true)]
-		public void OnActionExecuted_ValidatesEntity_IfNotOnExecuting(
-			bool onExecuting
-		)
+		public void OnActionExecuted_ValidatesEntity()
```
Comment (cailyncodes) [inline test/HedwigTests/Filters/ValidateEntityFilterTests.cs:18]: I think this is missing the `[Fact]` annotation

File: src/Hedwig/Models/Enrollment/EnrollmentProfile.cs [hunk 4]
```diff
@@ -8,6 +8,9 @@ public EnrollmentProfile()
 		{
 			CreateMap<Enrollment, EnrollmentSummaryDTO>();
 			CreateMap<EnrollmentSummaryDTO, Enrollment>();
+
+			CreateMap<Enrollment, CdcReportEnrollmentDTO>()
+				.ReverseMap();
```
[the remark this answers is not in the document]
Comment (jlhogan) [reply src/Hedwig/Models/Enrollment/EnrollmentProfile.cs:16]: yes and i made that change in open 939 PR :)

File: src/Hedwig/Models/Reports/OrganizationReport/OrganizationReportSummaryDTO.cs (hunk 2 above)
Comment (jlhogan) [reply src/Hedwig/Models/Reports/OrganizationReport/OrganizationReportSummaryDTO.cs:8]: I imagine this is eventually needed here as we may eventually show multiple types of org reports in the report summary page? but cuold remove for now and only add when needed. (that's the reason for this being "organizationReportSummary" and not "cdcReportSummary", but i could be convinced that that is also not useful)

File: src/Hedwig/Controllers/ReportsController.cs [hunk 5]
```diff
@@ -75,6 +77,7 @@ CdcReport report
 			if (report.Id != id) return BadRequest();
 			if (report.OrganizationId != orgId) return BadRequest();
 
+			_validator.Validate(report);
```
[the remark this answers is not in the document]
Comment (jlhogan) [reply src/Hedwig/Controllers/ReportsController.cs:80]: at the time i was confused abt how to update the filter setup to get this working, but i will try again :)

File: src/Hedwig/Models/Reports/OrganizationReport/OrganizationReportSummaryDTO.cs (hunk 2 above)
Comment (cailyncodes) [reply src/Hedwig/Models/Reports/OrganizationReport/OrganizationReportSummaryDTO.cs:8]: My thought is to remove for now (we can still call it OrganizationReportSummary) and add if/when needed. I think it's valuable to keep the complexity down (albeit true that this doesn't add much) until design comps occur for multiple reports use case

**Comment by jlhogan** [comment]
>General idea for consideration: Having an interface like HedwigDTO<T> where T is the class that the DTO is based on. 

I wonder if this simplifies the stuff i've done in other 939 PR by allowing us to only pass around the DTOs? would definitely love to pair on a walkthru!

**Comment by jlhogan** [comment]
> Not super necessary but it took me a little while to figure out what this DTO was actually trying to represent.

rethinking this ^, the entity the DTO represents should be clear from the folder the DTO is located in

**Pull request merged**